### ISSUE TITLE
Fix S4049 FP: Do not raise on methods with attributes that are not allowed on properties

### ISSUE THREAD

**Issue by Corniel** [SonarSource/sonar-dotnet#3140]
### Description
The warning S4049 (Properties should be preferred) is raised on methods that are decorated on attributes do not allowed on properties (such as web API controller methods).

### Repro steps
``` C#
public class S4049
{
    private static readonly Random rnd = new Random();

    [NoProperty]
    public int GetRandom() => rnd.Next(); // compliant, the attribute is not allowed on properties.

    [AttributeUsage(AttributeTargets.All ^ AttributeTargets.Property)]
    private class NoPropertyAttribute : Attribute { }
}
```
### Related information
* SonarC# Version 8.3.0.14607

**Comment by Corniel** [comment]
See https://github.com/SonarSource/sonar-dotnet/pull/3154

@pavel-mikula-sonarsource 
This is not fixed in the merged changes. A test is added that checks if the construct is noncompliant, but it should be compliant. So the test should be changed to be compliant, and then in the rule it should check if the method has any attribute that is not allowed on a property, if so, no warning should be generated.

**Comment by pavel-mikula-sonarsource** [comment]
Hi @Corniel ,

Thank you for reporting this issue, but we haven't fixed the issue yet. It is still open.

PR #3154 just adds reproducer into our unit tests to be aware of it and to be sure that it is confirmed and reproducible. It has to be marked as `// Noncompliant FP` for now to match current rule implementation.

When we fix this issue, we'll surely remove the `// Noncompliant` comment and close this issue as well.

**Comment by Corniel** [comment]
@pavel-mikula-sonarsource : Thanks, I was just not sure, if I understood the intent of the pull request correctly.
